File: MagicHax/android/app/src/main/java/me/nathanp/magiccreatures/model/Trait.java
package me.nathanp.magiccreatures.model;

import com.google.firebase.database.Exclude;

public class Trait extends Types.Typed {
    public String name;
    public int cost;
    public String description;

    @Exclude
    private boolean selected = false;

    public void setSelected(boolean b) {
        selected = b;
    }

    public boolean isSelected() {
        return selected;
    }
}

File: MagicHax/android/app/src/main/java/me/nathanp/magiccreatures/model/Trait.kt
package me.nathanp.magiccreatures.model

import me.nathanp.magiccreatures.util.Resources.Types.getTypesDrawable
import me.nathanp.magiccreatures.model.Types.Typed

data class Trait(
        var name: String = "",
        var cost: Int = 0,
        var description: String = ""
) : Typed() {
    val drawableId: Int
        get() = getTypesDrawable(*getTypeAsSortedInts())
}